airbyte-db/lib/src/main/java/io/airbyte/db/factory/DataSourceFactory.java

```diff
@@ -0,0 +1,251 @@
+   * @param port The port of the database.
+   * @param database The name of the database.
+   * @param driverClassName The fully qualified name of the JDBC driver class.
+   * @param connectionProperties Additional configuration properties for the underlying driver.
+   * @return The configured {@link DataSource}.
+   */
+  public static DataSource create(final String username,
+                                  final String password,
+                                  final String host,
+                                  final int port,
+                                  final String database,
+                                  final String driverClassName,
+                                  final Map<String, String> connectionProperties) {
+    return new DataSourceBuilder()
+        .withConnectionProperties(connectionProperties)
+        .withDatabase(database)
+        .withDriverClassName(driverClassName)
+        .withHost(host)
+        .withPort(port)
+        .withPassword(password)
+        .withUsername(username)
+        .build();
+  }
+
+  /**
+   * Convenience method that constructs a new {@link DataSource} for a PostgreSQL database using the
+   * provided configuration.
+   *
+   * @param username The username of the database user.
+   * @param password The password of the database user.
+   * @param host The host address of the database.
+   * @param port The port of the database.
+   * @param database The name of the database.
+   * @return The configured {@link DataSource}.
+   */
+  public static DataSource createPostgres(final String username,
+                                          final String password,
+                                          final String host,
+                                          final int port,
+                                          final String database) {
+    return new DataSourceBuilder()
+        .withDatabase(database)
+        .withDriverClassName("org.postgresql.Driver")
+        .withHost(host)
+        .withPort(port)
+        .withPassword(password)
+        .withUsername(username)
+        .build();
+  }
+
+  /**
+   * Builder class used to configure and construct {@link DataSource} instances.
+   */
+  private static class DataSourceBuilder {
+
+    private static final Map<String, String> JDBC_URL_FORMATS = Map.of("org.postgresql.Driver", "jdbc:postgresql://%s:%d/%s",
+        "com.amazon.redshift.jdbc.Driver", "jdbc:redshift://%s:%d/%s",
+        "com.mysql.cj.jdbc.Driver", "jdbc:mysql://%s:%d/%s",
+        "com.microsoft.sqlserver.jdbc.SQLServerDriver", "jdbc:sqlserver://%s:%d/%s",
+        "oracle.jdbc.OracleDriver", "jdbc:oracle:thin:@%s:%d:%s",
+        "ru.yandex.clickhouse.ClickHouseDriver", "jdbc:ch://%s:%d/%s",
+        "org.mariadb.jdbc.Driver", "jdbc:mariadb://%s:%d/%s");
+
+    private Map<String, String> connectionProperties = Map.of();
+    private String database;
+    private String driverClassName = "org.postgresql.Driver";
```

Comment:
Just curious, what is the reasoning for setting the default value of this property to postgres?

Reply:
Almost all of the `Database` objects that get created are for Postgresql -- especially those done via the `createPostgresqlDatabaseX` methods in the `Databases` class.  If we feel strongly enough, I can remove this and just throw an exception if the driver class name is not provided.

Reply:
I think we should limit ourself with what we need to use. It will be also quite hard to have an exhaustive list here.